src/test/java/seedu/trackermon/model/show/NameContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,75 @@
+package seedu.trackermon.model.show;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.trackermon.testutil.ShowBuilder;
+
+public class NameContainsKeywordsPredicateTest {
+
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("first");
+        List<String> secondPredicateKeywordList = Arrays.asList("first", "second");
+
+        NameContainsKeywordsPredicate firstPredicate = new NameContainsKeywordsPredicate(firstPredicateKeywordList);
+        NameContainsKeywordsPredicate secondPredicate = new NameContainsKeywordsPredicate(secondPredicateKeywordList);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        NameContainsKeywordsPredicate firstPredicateCopy = new NameContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different name -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void test_nameContainsKeywords_returnsTrue() {
+        // One keyword
+        NameContainsKeywordsPredicate predicate =
+            new NameContainsKeywordsPredicate(Collections.singletonList("Another"));
+        assertTrue(predicate.test(new ShowBuilder().withName("Another Erased").build()));
+
+        // Multiple keywords
+        predicate = new NameContainsKeywordsPredicate(Arrays.asList("Another", "Erased"));
+        assertTrue(predicate.test(new ShowBuilder().withName("Another Erased").build()));
+
+        // Only one matching keyword
+        predicate = new NameContainsKeywordsPredicate(Arrays.asList("Erased", "Oreimo"));
```

Comment:
I was just wondering if making the keywords into variables that you can reuse, instead of writing out the same string repeatedly would be better

Reply:
Noted, will do that.

Reply:
Actually, after trying to put variables into my test cases, I feel that there is no need to put into variables.

Below is just an example of how it would look like. 
![image](https://user-images.githubusercontent.com/65216105/162254187-5037dcb6-b3b6-4b2e-8885-c93882ff9378.png)

Reply:
If this was in a main class used for Trackermon I'd agree with @aweijun but since this is for testing, I think it'd be up to personal preference.